lib/awful/tag.lua

```diff
@@ -1350,11 +1352,35 @@ capi.tag.add_signal("property::urgent")
 
 capi.tag.add_signal("property::urgent_count")
 capi.tag.add_signal("property::volatile")
+capi.tag.add_signal("removal-pending")
 
 capi.screen.add_signal("tag::history::update")
 
 capi.screen.connect_signal("tag::history::update", tag.history.update)
 
+capi.screen.connect_signal("removed", function(s)
+    -- First give other code a chance to clean up
+    for _, t in pairs(s.tags) do
+        t:emit_signal("removal-pending")
```

Comment:
So you are saying:

1. I should rename `removal-pending` to `request::screen` (well, how about `request::new_screen`?)
2. I should do another loop afterwards with `for _, c in ipairs(s.clients) do c:emit_signal("request::tag") end` (how about `request::new_tag`?)

(And after the above, I keep the code that I already have)

Right?